core/wiki/macros/tree.tid

```diff
@@ -1,22 +1,15 @@
 title: $:/core/macros/tree
 tags: $:/tags/Macro
 
-\define leaf-link(full-title,chunk,separator: "/")
-<$link to=<<__full-title__>>><$text text=<<__chunk__>>/></$link>
-\end
-
-\define leaf-node(prefix,chunk)
+\define leaf-node(chunk,title)
 \whitespace trim
 <li>
-<$list filter="[<__prefix__>addsuffix<__chunk__>is[shadow]] [<__prefix__>addsuffix<__chunk__>is[tiddler]]" variable="full-title">
-<$list filter="[<full-title>removeprefix<__prefix__>]" variable="chunk">
-<span>{{$:/core/images/file}}</span>&#32;<$macrocall $name="leaf-link" full-title=<<full-title>> chunk=<<chunk>>/>
-</$list>
-</$list>
+<span>{{$:/core/images/file}}</span>&#32;
+<$link to=<<__title__>>><$text text=<<__chunk__>>/></$link>
 </li>
 \end
 
-\define branch-node(prefix,chunk,separator: "/")
+\define branch-node(prefix,chunk,field,separator: "/")
```

Comment:
For backwards compatibility reasons, we cannot change a macro signature. **So the new parameter has to go to the end of every macro call**, because TW allows unnamed parameter calls, where order is important. So you change the order of the parameters here, which will invalidate existing user code.

It's not nice, but it has to be that way

Reply:
Thank you, this was a very good point!